### PR TITLE
Implement deprecated method addAttributes in the Span and remove it from anywhere else.

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/census-instrumentation/opencensus-java/issues/624

### CODE REVIEW

File: api/src/main/java/io/opencensus/trace/Span.java [hunk 1]
```diff
@@ -109,7 +109,9 @@ public void putAttributes(Map<String, AttributeValue> attributes) {
    * @param attributes the attributes that will be added and associated with the {@code Span}.
    */
   @Deprecated
-  public abstract void addAttributes(Map<String, AttributeValue> attributes);
+  public void addAttributes(Map<String, AttributeValue> attributes) {
+    putAttributes(attributes);
```
Comment (sebright) [inline api/src/main/java/io/opencensus/trace/Span.java]: Should there be some kind of warning about the circularity, telling users that they must implement one of addAttributes or putAttributes?

Comment (bogdandrutu) [reply api/src/main/java/io/opencensus/trace/Span.java]: Done.

Comment (sebright) [reply api/src/main/java/io/opencensus/trace/Span.java]: Shouldn't it go in the Javadocs?

Comment (bogdandrutu) [reply api/src/main/java/io/opencensus/trace/Span.java]: I think this is not relevant for 99.9% of the users of our library so I would not put it in the javadoc. This is relevant only for other implementation which I don't really expect to exist.

**Pull request merged**